src/NuGet.Core/NuGet.CommandLine.XPlat/DeleteCommand.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Dnx.Runtime.Common.CommandLine;
+using NuGet.Commands;
+using NuGet.Configuration;
+using NuGet.Logging;
+using NuGet.Protocol.Core.Types;
+using NuGet.Protocol.Core.v3;
+using System.Threading;
+
+namespace NuGet.CommandLine.XPlat
+{
+    class DeleteCommand : Command
+    {
+        public DeleteCommand(CommandLineApplication app, Func<ILogger> getLogger)
+        {
+            app.Command("delete", delete =>
+            {
+                delete.Description = Strings.Delete_Description;
+
+                var source = delete.Option(
+                    "-s|--source <source>",
+                    Strings.Source_Description,
+                    CommandOptionType.SingleValue);
+
+                //TODO: communicate to Nuget team, this should go to a top level argument, and
+                //should not be handled by individual cmdlet
+                var nonInteractive = delete.Option(
+                    "--non-interactive <nonInteractive>",
+                    Strings.NonInteractive_Description,
+                    CommandOptionType.NoValue);
+
+                var argRoot = delete.Argument(
+                    "[root]",
+                    Strings.Delete_PackageIdAndVersion_Description,
+                    multipleValues: true);
+
+                delete.OnExecute(async () =>
+                {
+                    ILogger logger = getLogger();
+                    ISettings setting = Settings.LoadDefaultSettings(Path.GetFullPath("."),
+                                configFileName: null,
+                                machineWideSettings: null);
+                    PackageUpdateResource pushCommandResource = await GetPushCommandResource(source, setting);
+
+                    var packageId = argRoot.Values[0];
+                    var packageVersion = argRoot.Values.Count > 1 ? argRoot.Values[1] : string.Empty;
```

Comment:
Yes i will put in the validation. There are couple of interesting notes, 
For version string check, I tried out the latest nuget,exe 3.3, and appears it doesn't do it. So I will keep the same behavior, but let me know if you like see it happen.
If version is left off, at the client, the next argument like apikey will be taken as a version and the current nuget 3.3 starts prompt for credentials. Regarding whether server will go ahead to delete all versions, I am not clear, but i can clarify with John tomorrow. One this is clear, i will consolidate validations for how many arguments is expected.